### PR TITLE
Add default scheme if not provided for VERTA_HOST.

### PR DESCRIPTION
**Pull request by nhatsmrt**

### CODE REVIEW

File: client/scala/src/main/scala/ai/verta/client/ClientConnection.scala [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 case class ClientConnection(host: String, ignoreConnErr: Boolean = false, maxRetries: Int = 5, auth: ClientAuth = null)
 
 object ClientConnection {
-  def fromEnvironment(): ClientConnection = new ClientConnection(sys.env("VERTA_HOST"), auth = ClientAuth.fromEnvironment())
+  private val DefaultScheme = "https"
+
+  private def hasScheme(url: String) = (new URI(url)).getScheme() != null
+
+  private def urlWithScheme(url: String) =
+    if (hasScheme(url)) url else (new URI(DefaultScheme, url, null, null, null)).toString()
```
Comment (conradoverta) [inline client/scala/src/main/scala/ai/verta/client/ClientConnection.scala]: Could we just replace the scheme? This is making hte assumption that 3 parts of the URI are `null`, which seems like a stronger assumption than we have to make here.

**Review by conradoverta — APPROVED** [review]
Added one comment. LGTM otherwise.

File: client/scala/src/main/scala/ai/verta/client/ClientConnection.scala (hunk 1 above)
Comment (nhatsmrt) [reply client/scala/src/main/scala/ai/verta/client/ClientConnection.scala]: I think the most general solution is to pass the URL to the URI constructor, then use URI's method to extract the various components. Something like this:
```scala
val uri = new URI(url)

(new URI(Option(uri.getScheme()).getOrElse(DefaultScheme), uri.getAuthority(), uri.getPath(), uri.getQuery(), uri.getFragment())
```
Or one of the other constructors. Not sure which one is the best 🤔 
Screen Shot 2020-10-27 at 9 53 35 PM

Comment (nhatsmrt) [reply client/scala/src/main/scala/ai/verta/client/ClientConnection.scala]: UPDATE: ignore latest commit. That doesn't work for localhost:3000.

Comment (nhatsmrt) [reply client/scala/src/main/scala/ai/verta/client/ClientConnection.scala]: Actual neither is my original attempt =) It probably picks up localhost as scheme =)

**Comment by nhatsmrt** [comment]
OK third attempt: pretty much resort to hard-coding =)) This is too hacky for my taste.

File: client/scala/src/main/scala/ai/verta/client/ClientConnection.scala (hunk 1 above)
Comment (conradoverta) [reply client/scala/src/main/scala/ai/verta/client/ClientConnection.scala]: Localhost as scheme is crazy!

**Comment by conradoverta** [comment]
> OK third attempt: pretty much resort to hard-coding =)) This is too hacky for my taste.

I'm not sure we have a better option with the URI class behaving like it does.

**Review by conradoverta — APPROVED** [review]

**Pull request merged**